colour/io/luts/cinespace_csp.py

```diff
@@ -0,0 +1,394 @@
+# -*- coding: utf-8 -*-
+"""
+Cinespace .csp LUT Format Input / Output Utilities
+==================================================
+
+Defines *Cinespace* *.csp* *LUT* Format related input / output utilities
+objects.
+
+-   :func:`colour.io.write_LUT_Cinespace`
+
+References
+----------
+-   :cite: https://sourceforge.net/p/cinespacelutlib/code/HEAD/tree/trunk/
+"""
+
+from __future__ import division, unicode_literals
+
+import numpy as np
+import re
+import os
+
+from colour.constants import DEFAULT_FLOAT_DTYPE
+from colour.io.luts import LUT1D, LUT2D, LUT3D, LUTSequence
+from colour.utilities import tsplit, tstack
+
+__author__ = 'Colour Developers'
+__copyright__ = 'Copyright (C) 2013-2018 - Colour Developers'
+__license__ = 'New BSD License - http://opensource.org/licenses/BSD-3-Clause'
+__maintainer__ = 'Colour Developers'
+__email__ = 'colour-science@googlegroups.com'
+__status__ = 'Production'
+
+__all__ = ['read_LUT_Cinespace', 'write_LUT_Cinespace']
+
+
+def read_LUT_Cinespace(path):
+    """
+    Reads given *Cinespace* *.csp* *LUT* file.
+
+    Parameters
+    ----------
+    path : unicode
+        *LUT* path.
+
+    Returns
+    -------
+    LUT2D or LUT3D or LUTSequence
+        :class:`LUT2D` or :class:`LUT3D` or :class:`LUTSequence` class
+        instance.
+
+    Examples
+    --------
+    Reading a 2D *Cinespace* *.csp* *LUT*:
+
+    >>> path = os.path.join(
+    ...     os.path.dirname(__file__), 'tests', 'resources', 'cinespace',
+    ...     'ACES_Proxy_10_to_ACES.csp')
+    >>> print(read_LUT_Cinespace(path))
+    LUT2D - ACES Proxy 10 to ACES
+    -----------------------------
+    <BLANKLINE>
+    Dimensions : 2
+    Domain     : [[ 0.  0.  0.]
+                  [ 1.  1.  1.]]
+    Size       : (32, 3)
+
+    Reading a 3D *Cinespace* *.csp* *LUT*:
+
+    >>> path = os.path.join(
+    ...     os.path.dirname(__file__), 'tests', 'resources', 'cinespace',
+    ...     'ColourCorrect.csp')
+    >>> print(read_LUT_Cinespace(path))
+    LUT3D - Generated by Foundry::LUT
+    ---------------------------------
+    <BLANKLINE>
+    Dimensions : 3
+    Domain     : [[ 0.  0.  0.]
+                  [ 1.  1.  1.]]
+    Size       : (4, 4, 4, 3)
+    """
+
+    title = re.sub('_|-|\\.', ' ', os.path.splitext(os.path.basename(path))[0])
+    unity_range = np.array([[0., 0., 0.], [1., 1., 1.]])
+
+    def _parse_array(array):
+        """
+        Converts given string array to :class:`ndarray` class.
+        """
+
+        return np.array(list(map(DEFAULT_FLOAT_DTYPE, array)))
```

Comment:
I know I'm guilty of this but I will probably change `map` for a list comprehension in a later update.